clients/java/src/test/java/io/zeebe/client/workflow/PublishMessageTest.java

```diff
@@ -34,21 +35,23 @@
   @Test
   public void shouldPublishMessage() {
     // when
-    client
-        .newPublishMessageCommand()
-        .messageName("name")
-        .correlationKey("key")
-        .timeToLive(Duration.ofDays(1))
-        .messageId("theId")
-        .send()
-        .join();
+    final PublishMessageResponse response =
+        client
+            .newPublishMessageCommand()
+            .messageName("name")
+            .correlationKey("key")
+            .timeToLive(Duration.ofDays(1))
+            .messageId("theId")
+            .send()
+            .join();
 
     // then
     final PublishMessageRequest request = gatewayService.getLastRequest();
     assertThat(request.getName()).isEqualTo("name");
     assertThat(request.getCorrelationKey()).isEqualTo("key");
     assertThat(request.getMessageId()).isEqualTo("theId");
     assertThat(request.getTimeToLive()).isEqualTo(Duration.ofDays(1).toMillis());
+    assertThat(response.getMessageKey()).isZero();
```

Comment:
Let's check that we return the correct key. 

First, we need to instrument the gateway:
```
    // given
    final long messageKey = 123L;
    gatewayService.onPublishMessageRequest(messageKey);
```

When we can check the response:
```suggestion
    assertThat(response.getMessageKey()).isEqualTo(messageKey);
```

In `io.zeebe.client.util.RecordingGatewayService`, we need to add the new method to control the response:

```
  public void onPublishMessageRequest(final long key) {
    addRequestHandler(
        PublishMessageRequest.class,
        request -> PublishMessageResponse.newBuilder().setKey(key).build());
  }
```